framework/src/media/voice/SpeechDetector.cpp

```diff
@@ -401,6 +409,62 @@ void SpeechDetectorImpl::resetEndPointDetectorPtr(void)
 {
 	mEndPointDetector = nullptr;
 }
+
+void SpeechDetectorImpl::notifySync()
+{
+	std::unique_lock<std::mutex> lock(mCmdMtx);
+	mSyncCv.notify_one();
+}
+
+bool SpeechDetectorImpl::getKeywordBufferSize(uint32_t *bufferSize)
+{
+	std::unique_lock<std::mutex> lock(mCmdMtx);
+	SpeechDetectorWorker &sdw = SpeechDetectorWorker::getWorker();
+	bool returnValue;
+	sdw.enQueue([this, &bufferSize, &returnValue]() { this->_getKeywordBufferSize(bufferSize, returnValue); });
```

Comment:
below is startKeywordDetect
```
bool SpeechDetectorImpl::startKeywordDetect(void)
{
	if (mKeywordDetector == nullptr) {
		meddbg("KeywordDetector is not init\n");
		return false;
	}

	SpeechDetectorWorker &sdw = SpeechDetectorWorker::getWorker();
	sdw.enQueue(&KeywordDetector::startKeywordDetect, mKeywordDetector);

	medvdbg("Speech detector start KD done");
	return true;
}
```
Any specific reason you add notifySync & condition variable to speechdetector.cpp?
I can't understand this well

Reply:
But before framework get message 'KD', buffer already filled with data in driver.
Hence I think actually this can be done by sync call.